src/vmspawn/vmspawn.c

```diff
@@ -596,6 +616,68 @@ static int start_tpm(sd_bus *bus, const char *scope, const char *our_runtime_dir
         return 0;
 }
 
+static int find_initrd(char **ret_initrd) {
+        _cleanup_free_ char *s = NULL, *initrd = NULL;
+        char *c;
+
+        assert(ret_initrd);
+
+        /* if the kernel is an EFI image we don't need an initrd */
+        if (endswith(arg_kernel, ".efi")) {
+                ret_initrd = NULL;
+                return 0;
+        }
```

Comment:
This should use `inspect_kernel()` to figure out the type instead of checking the file extension

Reply:
To be fair I would drop this automatic initrd finding logic completely for now and just require the initrd to be specified explicitly.

Reply:
This certainly makes it a lot simpler 😂